de.dlr.sc.virsat.graphiti.test/src/de/dlr/sc/virsat/graphiti/diagram/BeanIndependenceSolverTest.java

```diff
@@ -76,91 +86,108 @@ protected void doExecute() {
-		ed.add(testDocument2);
-
-		ElementDefinition ed2 = new ElementDefinition(conceptEgscc);
-		StructuralElementInstance sei2 = ed2.getStructuralElementInstance();
-		
-		Document testDocument = new Document(concept);
-		
-		// Create The diagram
-		Diagram diagram = Graphiti.getPeCreateService().createDiagram("test", "testDiagram", true);
-		IFolder diagramFolder = testProject.getFolder("src/diagrams/");  
-		IFile diagramFile = diagramFolder.getFile("testDiagram" + "." + "test");  
-		URI uri = URI.createPlatformResourceURI(diagramFile.getFullPath().toString(), true);
-		DiagramHelper.createDiagram(uri, diagram, rs);
 
-		IDiagramTypeProvider dtp = GraphitiInternal.getEmfService().getDTPForDiagram(diagram);
-		GraphitiInternal.getEmfService().wireDTPToDiagram(diagram, dtp);
-		dtp.init(diagram, null);
-		BeanIndependenceSolver beanIndependenceSolver = new BeanIndependenceSolver(dtp);
-		String seiKey = beanIndependenceSolver.getKeyForBusinessObject(ed);
-		assertEquals(UUID, seiKey);
+		testDocument = new Document(concept);
+		testDocument2 = new Document(concept);
+		testDocument.getTypeInstance().setUuid(new VirSatUuid(DOCUMENTUUID));
 
-		seiKey = beanIndependenceSolver.getKeyForBusinessObject(ed);
-		assertEquals(UUID, seiKey);
+		IDiagramTypeProvider dtp = getDiagramTypeProvider();
+		beanIndependenceSolver = new BeanIndependenceSolver(dtp);
 
-		testDocument.getTypeInstance().setUuid(new VirSatUuid(DOCUMENTUUID));
-		String caKey = beanIndependenceSolver.getKeyForBusinessObject(testDocument);
-		assertEquals(DOCUMENTUUID, caKey);
+		ed1 = new ElementDefinition(conceptEgscc);
+		ed2 = new ElementDefinition(conceptEgscc);
+		sei1 = ed1.getStructuralElementInstance();
+		sei2 = ed2.getStructuralElementInstance();
+		sei1.setUuid(new VirSatUuid(UUID));
+		ed1.add(testDocument2);
+		getAndAddStructuralElementInstance(sei1);
 
-		caKey = beanIndependenceSolver.getKeyForBusinessObject(testDocument);
-		assertEquals(DOCUMENTUUID, caKey);
+		command = CreateAddStructuralElementInstanceCommand.create(editingDomain, repository, ed1.getStructuralElementInstance());
+	}
 
-		Command command = CreateAddStructuralElementInstanceCommand.create(editingDomain, repository, ed.getStructuralElementInstance());
-		editingDomain.getCommandStack().execute(command);
+	@Test
+	public void testSolveIndependenceForSei() throws CoreException, IOException {
+		String seiKey = beanIndependenceSolver.getKeyForBusinessObject(ed1);
+		assertEquals(UUID, seiKey);
 
-		editingDomain.getVirSatCommandStack().execute(new RecordingCommand(editingDomain) {
-			@Override
-			protected void doExecute() {
-				rs.getAndAddStructuralElementInstanceResource(sei2);
-			}
-		});
+		editingDomain.getCommandStack().execute(command);
+		getAndAddStructuralElementInstance(sei2);
 
-		Resource resEd2 = rs.getStructuralElementInstanceResource(ed2.getStructuralElementInstance());
-		resEd2.getContents().add(ed2.getStructuralElementInstance());
+		addStructuralElementToResource(ed2);

```

Comment:
Please improve comments, it is ahrd to understand which uuid is the SEI and which one is the BO and so on. mabye consider using the comments in the assert statements as well. This makes it easier to spot the exact place and reason once a test case is failing.